### PR TITLE
Add HTTP logging

### LINKED ISSUE #2960: Add in-app logging of HTTP requests
## Describe your issue

It would be nice to debug HTTP requests easely inside of the app.

## Proposed solutions

 - Adding a log into F10 debug popup
 - more logging
 - custom channel (kinda like `$$$`)

### PR DESCRIPTION
**Pull request by Mm2PL**
Pull request checklist:

- [x] `CHANGELOG.md` was updated, if applicable

# Description

Adds logging for completed / timed out / cancelled / errored HTTP requests. Closes #2960

### CODE REVIEW

File: src/common/NetworkPrivate.cpp
Comment (zneix) [inline src/common/NetworkPrivate.cpp:209]: Here, and in all other places with `QString::arg` methods we should use multi-arg instead, see https://github.com/KDE/clazy/blob/master/docs/checks/README-qstring-arg.

(I'll make a PR fixing all those other places with this in codebase later.

**Review by zneix — CHANGES_REQUESTED** [review]

File: src/common/NetworkPrivate.cpp
Comment (Mm2PL) [reply src/common/NetworkPrivate.cpp]: 1c9e6f2 (will resolve when and if it compiles)

**Comment by Mm2PL** [comment]
multi arg broke it, feelsgoodman

**Review by pajlada — APPROVED** [review]

File: src/common/NetworkCommon.hpp
Comment (zneix) [inline src/common/NetworkCommon.hpp:32]: This would make diffs cleaner in the future if we're ever to add anything here. It probably won't happen, but it seems cleaner to have that comma at the end imho.
```suggestion
    "PATCH",   //
```

**Review by zneix — CHANGES_REQUESTED** [review]
I'll test the changes right now and approve if it seems fine. Got one little comment as well.

**Comment by zneix** [comment]
I'd personally change order of things in logs, e.g. `GET 200 https://supalonglink.com/aaaaaaaaaaaaaaaaaaaaaaaa` makes it easier to look at the requests than current solution - `GET https://supalonglink.com/aaaaaaaaaaaaaaaaaaaaaaaa 200`

**Comment by zneix** [comment]
Functionality itself looks good and :+1: for implementing it, however I don't like the name of logging category - `chatterinoHTTP`. All other logging category names follow the format `chatterino` + `Name`, where `Name` is always starting uppercased, but rest is lowercased.
Maybe I'm ultra annoying and picky for no reason, but I just feel like it should follow the already established format and use `chatterinoHttp`.

File: CHANGELOG.md
Comment (pajlada) [inline CHANGELOG.md:36]:
```suggestion
```

**Review by pajlada — CHANGES_REQUESTED** [review]
fix the changelog stuff then I'm good to merge
`chatterinoHTTP` > `chatterinoHttp` 
image

**Review by zneix — APPROVED** [review]
Really dislike `chatterinoHTTP`, but given the based emote I'm good with merging this in. Waiting for pajlada's approval.

**Review by pajlada — APPROVED** [review]

**Pull request merged**
